### PR TITLE
Wazo 813 voicemail greeting api

### PR DESCRIPTION
**Pull request by nballas-wazo**

### CODE REVIEW

File: debian/patches/wazo_ari_expose_voicemail_functions [hunk 1]
```diff
@@ -524,3 +1262,4596 @@ Index: asterisk-16.4.0/rest-api/resources.json
...
+-                                if (!res)
+-                                        res = ast_play_and_wait(chan, "digits/1");
+-                                if (!res)
+-                                        res = ast_play_and_wait(chan, "jp-wo");
+-                                if (!res)
+-                                        res = ast_play_and_wait(chan, "silence/1");
+-                                res = ast_play_and_wait(chan, "vm-opts-full");
+-                        }
+-                        /* Logic:
+-                         * If the current mess
```
Comment (sduthil) [inline debian/patches/wazo_ari_expose_voicemail_functions]: Shouldn't we check the return value of fwrite?

Comment (sduthil) [inline debian/patches/wazo_ari_expose_voicemail_functions]: Please don't use uppercase for error messages.

Comment (sduthil) [inline debian/patches/wazo_ari_expose_voicemail_functions]: We should check if the file exists with `ast_fileexists`

Comment (sduthil) [inline debian/patches/wazo_ari_expose_voicemail_functions]: We do not need to append a NULL character after binary data, do we?

Comment (sduthil) [inline debian/patches/wazo_ari_expose_voicemail_functions]: We should use `setup_vm_path` for this.

Comment (sduthil) [inline debian/patches/wazo_ari_expose_voicemail_functions]: Maybe rename to `get_vm_greeting_path` ?

Comment (sduthil) [inline debian/patches/wazo_ari_expose_voicemail_functions]: Unnecessary whitespaces here

File: debian/patches/wazo_ari_expose_voicemail_functions [hunk 2]
```diff
@@ -103,15 +101,198 @@ Index: asterisk-16.4.0/res/ari/resource_wazo.c
...
++
++	const char *greeting = args->greeting;
++	if(!greeting) {
++		ast_ari_response_error(response, 400, "Invalid argument", "Invalid voicemail greeting: BUSY or UNAVAILABLE or NAME");
++		return;
++	}
++
++	int length = strlen(encoded_data);
++	unsigned char *decoded_data = ast_calloc(length + 2, sizeof(char)); /*account for NULL character*/
++	int n = ast_base64decode(decoded_data, encoded_data, length - 1);
```
[the remark this answers is not in the document]
Comment (nballas-wazo) [reply debian/patches/wazo_ari_expose_voicemail_functions]: Documentation recommends `length` being `sizeof(destination) - 1`

**Pull request merged**